src/Microsoft.TestPlatform.PlatformAbstractions/common/System/ProcessHelper.cs

```diff
@@ -131,15 +131,18 @@ public bool TryGetExitCode(object process, out int exitCode)
         }
 
         /// <inheritdoc/>
-        public void SetExitCallback(int processId, Action callbackAction)
+        public void SetExitCallback(int processId, Action<object> callbackAction)
         {
-            var process = Process.GetProcessById(processId);
-
-            process.EnableRaisingEvents = true;
-            process.Exited += (sender, args) =>
+            try
+            {
+                var process = Process.GetProcessById(processId);
+                process.EnableRaisingEvents = true;
+                process.Exited += (sender, args) => callbackAction.Invoke(process);
```

Comment:
In line 140, try `sender as Process` within Invoke (reference: https://github.com/smadala/vstest/blob/0848eafca059e3e49790a8a91ea2b9919970de83/src/Microsoft.TestPlatform.PlatformAbstractions/common/System/ProcessHelper.cs#L55)